lib/src/country_flags.dart

```diff
@@ -92,4 +94,47 @@ class CountryFlag extends StatelessWidget {
       ),
     );
   }
+
+  /// Build the flag widget according to the specified shape.
+  Widget _buildFlagWidget() {
+    switch (shape) {
+      case Shape.circle:
+        return _buildCircularFlag();
+      case Shape.rectangle:
+        return _buildRectangularFlag();
+    }
+  }
+
+  /// Build the circular flag widget.
+  Widget _buildCircularFlag() {
+    return ClipOval(
+      child: ScalableImageWidget.fromSISource(
```

Comment:
Let's transform this `ScalableImageWidget.fromSISource` into a custom widget so we can reuse it in the `_buildCircularFlag()` method and the `_buildRectangularFlag()`.